pkg/machinery/client/config/path.go

```diff
@@ -5,43 +5,92 @@
 package config
 
 import (
+	"fmt"
 	"os"
 	"path/filepath"
 
 	"github.com/talos-systems/talos/pkg/machinery/constants"
 )
 
+// Path represents a path to a configuration file.
+type Path struct {
+	// Path is the filesystem path of the config.
+	Path string
+	// AllowWrite is true if the path is allowed to be written.
+	AllowWrite bool
```

Comment:
nit: looks better?

```suggestion
	Writeable bool
```

Reply:
`Writeable` sounded to me like a filesystem permissions thing, while actually it is our policy and has nothing to do with FS. That's why I chose the name. But no strong opinions, can change.

Reply:
`WriteAllowed`, sounds good, updated it.